webapp/views/App/views/Home/SurveyCreate/store/actions/useOnCreate.js

```diff
@@ -0,0 +1,27 @@
+import axios from 'axios'
+import { useDispatch } from 'react-redux'
+
+import { SurveyActions } from '@webapp/store/survey'
+
+export const useOnCreate = ({ newSurvey, setNewSurvey }) => {
+  const dispatch = useDispatch()
+
+  return () => {
+    ;(async () => {
+      const { name, label, lang } = newSurvey
+
+      const {
+        data: { survey, validation },
+      } = await axios.post('/api/survey', { name, label, lang })
+
+      if (survey) {
+        dispatch({ type: SurveyActions.surveyCreate, survey })
```

Comment:
wouldn't it be better to have the actions directly in survey actions? 
e.g.
```dispatch(SurveyActions.createSurvey({survey})```

Reply:
good catch